packages/data-flow/src/orchestrator.ts

```diff
@@ -113,16 +113,25 @@ export class Orchestrator {
                 await this.eventsRegistry.saveLastProcessedEvent(this.chainId, event);
 
                 event = await this.enhanceStrategyId(event);
-                if (event.contractName === "Strategy" && "strategyId" in event) {
+                if (this.isPoolCreated(event)) {
+                    // we save the strategy id with handled = false because we don't know if the strategy is handled yet
```

Comment:
why don't we check with `if (!existsHandler(event.strategyId)) {` to determine if it is handleable or not ?

Lets have a brief chat offline to define this :)